modules/katello_repository.py

```diff
@@ -130,8 +140,10 @@ def repository(self, name, content_type, product, organization, url=None):
 
         repository = repository[0] if len(repository) == 1 else None
 
-        if repository and (repository.name != name or repository.url != url):
-            repository = self._entities.Repository(self._server, name=name, id=repository.id, url=url)
+        # Sadly, nailgun does not handle 'download_policy' as we would like it.
```

Comment:
Does this imply it currently only works on creation? Do we have an issue open to nailgun for this?

Reply:
If i remember correctly, it is write-only. So creating and updating works, but reading (which is important for 'check_mode' and nice in any case) does not. Haven't had the time to get this to nailgun yet.

Reply:
The only thing @ nailgun i can find right now is not in repositories, but in smart-proxy. Reinvestigating this right now.

Reply:
Nailgun was not to blame here. Sorry!
A missing 'read()' was.

Reply:
Awesome!